python/hyperon/stdlib.py

```diff
@@ -52,6 +53,33 @@ def bool_ops():
         r"not": notAtom
     }
 
+@register_atoms
+def text_ops():
+    """Add text operators
+
+    repr: convert Atom to string.
+    parse: convert String to Atom.
+    stringToChars: convert String to tuple of Char.
+    charsToString: convert tuple of Char to String.
+
+    see test_stdlib.py for examples.
+
+    """
+    reprAtom = OperationAtom('repr', lambda a: [ValueAtom(repr(a))],
+                             ['Atom', 'String'], unwrap=False)
+    parseAtom = OperationAtom('parse', lambda s: [ValueAtom(SExprParser(str(s)[1:-1]).parse(Tokenizer()))],
+                              ['String', 'Atom'], unwrap=False)
+    stringToCharsAtom = OperationAtom('stringToChars', lambda s: [ValueAtom(E(*[ValueAtom(Char(c)) for c in str(s)[1:-1]]))],
+                                      ['String', 'Atom'], unwrap=False)
```

Comment:
We could possibly use `Expression` instead of `Atom` here.

Reply:
What is the difference between `Atom` and `Expression` (I tried to understand the difference by reviewing code and examples but couldn't)?  And why would `Expression` be better than `Atom` in that context?

Reply:
For `(: f (-> Expression Atom))`, `(f A)` should not type check, while `(f (A B))` should.
JIC, I put my comment under definition of `charsToStringAtom`, where `a.get_children()` is used (but it was put under `stringToCharsAtom`). Using `Expression` type should prevent `get_children` to be called in case, when non-expression atom is passed as an argument.

Reply:
Hmm... It doesn't seem to work now, after all. `car-atom` throws a error that it expects `Expression`, but it is not due its type, but it's hardcoded in its Rust implementation. However, `(f A)` gets reduced, if, say, `(= (f $x) $x)`.

Reply:
Oh, so `Atom = Symbol + Expression`, correct?

Reply:
`Symbol`, `Expression`, `GroundedAtom` are subtypes for `Atom`.

Reply:
OK, so should I conclude that (as of today)
1. `Atom = Symbol + Expression + GroundedAtom`
2. None of `Symbol`, `Expression` and `GroundedAtom` are subtypes of one another?

Reply:
yes